### PR TITLE
chore: Merged main branch into modular-sdk

### PR DESCRIPTION
**Pull request by hiranya911**
Synching modular-sdk branch with the latest head of the main branch.

@lahirumaramba @yuchenshi @rsgowman can I ask you to take a peek, and do a sanity check to make sure the changes you've worked on over the past couple of months are correctly integrated into this branch? All unit and integration tests are green (although I did find one bad merge related to emulator-based integration tests, which I've fixed since).

### CODE REVIEW

**Review by yuchenshi — APPROVED** [review]
LGTM for auth changes

File: src/auth/token-verifier.ts [hunk 1]
```diff
@@ -177,15 +179,20 @@ export interface DecodedIdToken {
...
 const CLIENT_CERT_URL = 'https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com';
 
 // URL containing the public keys for Firebase session cookies. This will be updated to a different URL soon.
 const SESSION_COOKIE_CERT_URL = 'https://www.googleapis.com/identitytoolkit/v3/relyingparty/publicKeys';
 
+/**
+ * User facing token information related to the Firebase ID token.
+ *
+ * @internal
+ */
```
Comment (lahirumaramba) [inline src/auth/token-verifier.ts:193]: This looks off. This comment shouldn't be here.

File: src/utils/jwt.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+export interface SignatureVerifier {
+  verify(token: string): Promise<void>;
+}
+
+interface KeyFetcher {
+  fetchPublicKeys(): Promise<{ [key: string]: string }>;
+}
+
+/**
+ * Class to fetch public keys from a client certificates URL.
```
Comment (lahirumaramba) [inline src/utils/jwt.ts:48]: Unrelated to this PR, but do these comments now need `@internal` annotations?

**Review by lahirumaramba — APPROVED** [review]
Thanks Hiranya!
Reviewed `nightly build notifications` and `jwt verification` changes. Looks good to me except for one glitch. Left comments.

File: src/utils/jwt.ts (hunk 2 above)
Comment (hiranya911) [reply src/utils/jwt.ts:48]: Most of the time it's not required. But as a best practice I think we will ask developers add this annotation to any APIs that are exported from their residing modules, but are not part of the public API. This makes sure the generated d.ts files remain small.

File: src/auth/token-verifier.ts (hunk 1 above)
Comment (hiranya911) [reply src/auth/token-verifier.ts]: Thanks. That looks like another bad merge. Removed.

**Comment by hiranya911** [comment]
Hi @rsgowman. Can you also take a quick look, and see your recent changes have been cleanly merged into this branch?

**Review by rsgowman — APPROVED** [review]
lgtm

**Pull request merged**